Automattic/pocket-casts-android
# Playback 2023: Tracks

| 📘 Part of: #1463 |
|:---:|

## Description
Adds event when showing the stories upsell

## Testing Instructions

- Enable EoY Year for release build
- Release build
- License test account

1. Make sure you're logged in to a free account that has a few episodes listened to this year and last year
2. Go to Profile 
3. Open stories
4. Go to the year over year story
5. ✅ Check that `end_of_year_upsell_shown` is tracked and `end_of_year_story_shown properties {"story": "year_over_year"}` is not
6. Tap upsell button
7. Check that `plus_promotion_shown properties {"flow": "plus_upsell, "source": "end_of_year"}` is shown. 
8. Purchase
9. ✅ Check that `end_of_year_story_shown ["story": "year_over_year"]` is shown
10. Go to next story
11. ✅ Check that `end_of_year_story_shown ["story": "completion_rate"]` is shown


At step 7. iOS displays 
`plus_promotion_shown ["flow": "endOfYearUpsell", "source": "endOfYear"]`
I am not sure if we need to specify a separate flow for end of year, given that we also display end of year source. Lmkwyt.

## Checklist
 
#### I have tested any UI changes...
